source/features/view-markdown-source.tsx

```diff
@@ -32,15 +32,15 @@ This acts as an auto-discarded cache without globals, timers, etc.
 It should also work clicks on buttons sooner than the page loads.
 */
 async function showSource() {
-	const sourceButton = select('.rgh-md-source');
-	const renderedButton = select('.rgh-md-rendered');
+	const sourceButton = select('.rgh-md-source')!;
+	const renderedButton = select('.rgh-md-rendered')!;
 
 	document.dispatchEvent(new CustomEvent('pjax:start')); // Show loading bar
 
-	const source = sourceButton[linkedDom] || fetchSource();
-	const rendered = renderedButton[linkedDom] || select('.blob.instapaper_body');
-	sourceButton[linkedDom] = source;
-	renderedButton[linkedDom] = rendered;
+	const source = (sourceButton as any)[linkedDom] as Element || fetchSource();
```

Comment:
I don't like what I've done here, but I can't think of another option at the moment. Initially my reflex was to extend the interface for `HTMLElement`.

```javascript
interface HTMLElement {
   [key: string]: Element;
}
```

but the key is a `Symbol` so that doesn't fly. Definitely open to suggestions.

Reply:
Alternatively we can store them in a WeakMap. I’m not sure why I didn’t do it in the first place.

Reply:
I was trying to avoid modifying any code while adding the types, but that approach makes more sense. It's also clearer vs. something just dangling off a DOM element.